common/src/main/java/com/revenuecat/purchases/common/HTTPClient.kt

```diff
@@ -173,7 +173,10 @@ class HTTPClient(
             throw IOException(NetworkStrings.HTTP_RESPONSE_PAYLOAD_NULL)
         }
 
-        val verificationResult = if (shouldSignResponse && nonce != null) {
+        val verificationResult = if (shouldSignResponse &&
+            nonce != null &&
+            RCHTTPStatusCodes.isSuccessful(responseCode)
```

Comment:
This is skipping it for 304s too, is that intended?

Reply:
Hmm this is only filtering 4xx/5xx error codes actually. Lmk if I missed something though

Reply:
Oh ok, I was just looking at how `isSuccessful` is defined in iOS not in Android 👍🏻